packages/data-driver-postgres/src/index.ts

```diff
@@ -31,13 +33,28 @@ export default class DataDriverPostgres implements DataDriver {
 		await this.#pool.end();
 	}
 
-	async query(query: AbstractQuery): Promise<Readable> {
+	async query(query: AbstractQuery): Promise<ReadableStream> {
```

Comment:
I figured this function would be nice to unit test as well. some mocking would be involved but should be doable. We could give an abstract query and check if it returned the expected nested result object for a mocked database response.

Reply:
done